### PR TITLE
storage: fix merge skew

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8648)

### CODE REVIEW

File: storage/raft_transport.go [hunk 1]
```diff
@@ -394,9 +394,13 @@ func (t *RaftTransport) SendAsync(req *RaftMessageRequest) bool {
 
 // SendSync sends a raft message and waits for an acknowledgement.
 func (t *RaftTransport) SendSync(ctx context.Context, req *RaftMessageRequest) error {
-	conn := t.getNodeConn(req.ToReplica.NodeID)
-	if conn == nil {
-		return errors.Errorf("unable to get connection for node %s", req.ToReplica.NodeID)
+	addr, err := t.resolver(req.ToReplica.NodeID)
```
Comment (spencerkimball) [inline storage/raft_transport.go]: This is a fix, but not what I'm intending. I'd like to reuse the breaker the previous change introduced.

File: storage/client_raft_test.go [hunk 2]
```diff
@@ -2330,9 +2330,11 @@ func TestFailedPreemptiveSnapshot(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	const expErr = "aborted due to failed preemptive snapshot: unknown peer 3"
```
Comment (spencerkimball) [inline storage/client_raft_test.go]: Is this breakage caused by the merge?

Comment (tamird) [inline storage/client_raft_test.go]: Yes, the previous wording was removed in your change.

**Comment by tamird** [comment]
@spencerkimball is going to fix this in a different way.

**Pull request closed without merging**